contactaccessrights/CRM/Contactaccessrights/Test/Fabricator/Rights.php

```diff
@@ -0,0 +1,41 @@
+<?php
+
+/**
+ * Fabricates rights for tests.
+ *
+ * @author Camilo Rodriguez - camilo@compucorp.co.uk
+ */
+abstract class CRM_Contactaccessrights_Test_Fabricator_Rights {
+	
+	/**
+	 * Fabricates Rights, given specific values forcontact_id, entity_id and 
+	 * entity_type.
+	 * 
+	 * @param array $params
+	 *   Associative array of parameters to be used to store rights infrmation in 
+	 *   database.
+	 * @return array
+	 *   Holds values stored in datbase
+	 * @throws Exception
+	 * @throws CiviCRM_API3_Exception
+	 */
+	public static function fabricate($params) {
+    if (!isset($params['contact_id']) || !isset($params['entity_id']) 
+			|| !isset($params['entity_type'])) {
+      throw new Exception('Specify contact_id, entity_id and entity_type [' . json_encode($params) . ']');
+    }
+
+		try {
+			$result = civicrm_api3('Rights', 'create', [
+				'sequential' => 1,
+				'contact_id' => $params['contact_id'],
+				'entity_id' => $params['entity_id'],
+				'entity_type' => $params['entity_type']
+			]);
+		} catch (CiviCRM_API3_Exception $e) {
+			throw new Exception("Error creating Rights! Params: " . serialize($params), 0, $e);

```

Comment:
I don't think we need such a specific exception here. Fabricator will only be used in tests. If a CiviCRM_API3_Exception is thrown, that specific test will fail and then it will be easy to figure out where was the fail and which param were passed to the API